### PR TITLE
Added gradle convention plugins

### PR DESCRIPTION
**Pull request by Kantis**

### CODE REVIEW

File: buildSrc/src/main/kotlin/kotest-js-conventions.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+plugins {
+   id("kotlin-conventions")
+}
+
+kotlin {
+   targets {
+      js(BOTH) {
```
Comment (Kantis) [inline buildSrc/src/main/kotlin/kotest-js-conventions.gradle.kts]: We sometimes used to target `js(IR)` and sometimes `js(BOTH)`. Are we ready to go to IR only?

File: buildSrc/build.gradle.kts [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
 repositories {
-   mavenCentral()
+   gradlePluginPortal()
 }
 
-plugins {
-   `kotlin-dsl`
+dependencies {
+   implementation("com.adarshr:gradle-test-logger-plugin:3.2.0")
+   implementation("org.jetbrains.kotlin:kotlin-gradle-plugin:1.6.21")
```
Comment (Kantis) [inline buildSrc/build.gradle.kts]: TODO: Should pull kotlin version from version catalog..

File: buildSrc/src/main/kotlin/kotest-multiplatform-conventions.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      watchosArm32()
+      watchosArm64()
+      watchosX86()
+      watchosX64()
+      watchosSimulatorArm64()
+
+      iosX64()
+      iosArm64()
+      iosArm32()
+      iosSimulatorArm64()
```
Comment (Kantis) [inline buildSrc/src/main/kotlin/kotest-multiplatform-conventions.gradle.kts]: TODO: Option to configure targets, all multiplatform targets should still be default.

Comment (Kantis) [reply buildSrc/src/main/kotlin/kotest-multiplatform-conventions.gradle.kts]: Having a hard time figuring this one out.. perhaps hard to solve with buildSrc plugins, might be more suited for a proper internal kotest gradle plugin..

File: buildSrc/src/main/kotlin/kotest-js-conventions.gradle.kts (hunk 1 above)
Comment (sksamuel) [reply buildSrc/src/main/kotlin/kotest-js-conventions.gradle.kts]: I guess once 1.7 comes out we could ditch BOTH ?

**Review by sksamuel — APPROVED** [review]

**Pull request merged**